website/docs/guides/data_solution_extension.html.markdown

```diff
@@ -0,0 +1,317 @@
+renders these values as an input form. It is not that trivial to provide such option for CLI
+applications, like Terraform. Terraform provider VCD attempts to provide as much convenience as
+possible by providing dynamic input validation in
+[`vcd_solution_add_on_instance`](/providers/vmware/vcd/latest/docs/resources/solution_add_on_instance)
+resource.
+
+The way it works is it will read provided input schema of a Solution Add-On and dynamically validate
+(during `apply` operation) if the provided inputs match requested ones. If they don't - it will
+print all the missing inputs with an error message that contains details for each of the missing
+fields (example below).
+
+In the printed error message, each field has an `IsDelete` flag which defines whether it should be
+specified in `input` or `delete_input` value in
+[`vcd_solution_add_on_instance`](/providers/vmware/vcd/latest/docs/resources/solution_add_on_instance)
+resource.
+
+All fields also have `Required` flag which hints if they are mandatory or not. By default,
+[`vcd_solution_add_on_instance`](/providers/vmware/vcd/latest/docs/resources/solution_add_on_instance)
+resource requires providing all `input` and `delete_input` values. If one doesn't want to specify
+some of the non-mandatory fields, it is possible to disable validation for non required fields by
+setting `validate_only_required_inputs = true`.
+
+**Note.** The `delete_input` fields are validated during removal (`destroy` operation). It may occur
+that these values have to be adjusted during removal phase - for that reason it is safe to update
+`delete_input`, perform `apply` (update operation will be no-op) and then retry `destroy` operation.
+
+
+```shell
+...
+╷
+│ Error: dynamic creation input field validation error: 
+│ -----------------
+│ Field: delete-previous-uiplugin-versions
+│ Title: Delete Previous UI Plugin Versions
+│ Type: Boolean
+│ Required: true
+│ IsDelete: false
+│ Description: If setting true, the installation will delete all previous versions of this ui plugin. If setting false, the installation will just disable previous versions
+│ Default: false
+│ -----------------
+│
+│ 
+│ ERROR: Missing fields 'delete-previous-uiplugin-versions' for Solution Add-On 'vmware.ds-1.4.0-23376809'
+│ 
+│   with vcd_solution_add_on_instance.dse14,
+│   on vcd.TestAccSolutionAddonInstanceAndPublishingstep1.tf line 96, in resource "vcd_solution_add_on_instance" "dse14":
+│   96: resource "vcd_solution_add_on_instance" "dse14" {
+...
+```
+
+## Publishing a Solution Add-On Instance
+
+The last step for making the Solution Add-On available is publishing it to tenants.
+
+```hcl
+resource "vcd_solution_add_on_instance_publish" "public" {
+  add_on_instance_id     = vcd_solution_add_on_instance.dse14.id
+  org_ids                = [data.vcd_org.recipient.id]
+  publish_to_all_tenants = false
+}
+```
+
+**Important!** Client must logout and login back to VCD so that newly published Solution Add-On can
```

Comment:
Should this have a `~>` ?

Reply:
Makes sense, I forgot about that.Changed.